### PR TITLE
[Fix](Job)The INSERT execution failed, but the task record status is marked as successful.

### PR DESCRIPTION
**Pull request by CalvinKirs**
### What problem does this PR solve?

After a job execution is completed, we need to verify whether the QueryState is normal.
Currently, we rely on exception handling to determine success. However, in certain cases, such as execution timeouts, exceptions may not be captured. As a result, the status is incorrectly marked as successful.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Manual test (add detailed scripts or steps below)
```
select * from tasks('type'='insert') ;


+----------------+----------------+---------+-------------------------------+---------+----------------------------------------------+---------------------+---------------------+------------+-------------+---------------+------+
| TaskId         | JobId          | JobName | Label                         | Status  | ErrorMsg                                     | CreateTime          | StartTime           | FinishTime | TrackingUrl | LoadStatistic | User |
+----------------+----------------+---------+-------------------------------+---------+----------------------------------------------+---------------------+---------------------+------------+-------------+---------------+------+
| 53767413936871 | 53748267972932 | test    | 53748267972932_53767413936871 | RUNNING |                                              | 2024-11-19 21:39:46 | 2024-11-19 21:39:46 |            |             |               | root |
| 53758617801828 | 53748267972932 | test    | 53748267972932_53758617801828 | FAILED  | errCode = 2, detailMessage = Execute timeout | 2024-11-19 21:39:37 | 2024-11-19 21:39:37 |            |             |               | root |

```

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by CalvinKirs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertTask.java [hunk 1]
```diff
@@ -195,10 +196,13 @@ public void run() throws JobException {
...
             }
             command.runWithUpdateInfo(ctx, stmtExecutor, loadStatistic);
+            if (ctx.getState().getStateType() != QueryState.MysqlStateType.OK) {
+                throw new JobException(ctx.getState().getErrorMessage());
+            }
         } catch (Exception e) {
             log.warn("execute insert task error, job id is {}, task id is {},sql is {}", getJobId(),
                     getTaskId(), sql, e);
-            throw new JobException(e);
+            throw new JobException(e.getMessage());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertTask.java:205]: We can use `Util.getRootCauseMessage()` to get the root cause, for better debug

**Comment by CalvinKirs** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**